ember-flatpickr/src/components/ember-flatpickr.ts

```diff
@@ -133,8 +134,10 @@ export default class EmberFlatpickr extends Component<EmberFlatpickrArgs> {
       Object.entries(rest).filter((entry) => entry[1] !== undefined),
     );
 
-    if (typeof this.args.locale === 'string' && this.args.locale !== "en") {
-      await import(`flatpickr/dist/l10n/${this.args.locale}.js`);
+    if (typeof this.args.locale === 'string' && this.args.locale !== 'en') {
+      await waitForPromise(
```

Comment:
@ijlee2 can you please explain what this is doing? I would have thought awaiting the import would be enough.

Reply:
`await import` is an asynchronous task that is pure JavaScript, so Ember tests can't know when translations are available (when the component is in a settled state), without using `waitFor()` or `waitUntil()` from `@ember/test-helpers`, for example.

I think the tests in this repo might have been passing, because the time to dynamically load the French translations has been short. On the other hand, tests in my production project have sometimes failed, because it can take a while to render a page.

By wrapping the dynamic import with `waitForPromise()`, projects that consume `ember-flatpickr` don't need to write additional code such as `waitFor()` or `waitUntil()` in their tests.

```diff
module('Integration | Component | appointment-form/date', function (hooks) {
  setupRenderingTest(hooks);

  test('We can show the current date', async function (assert) {
    await render(hbs`
      <AppointmentForm::Date
        @date={{this.date}}
      />
    `);
-     await waitUntilDateRenders(); // a test helper that uses waitUntil()

    assert.dom('[data-test-field="Date"]').hasValue('06.08.2024');
  });

  test('We can update the date', async function (assert) {
    await render(hbs`
      <AppointmentForm::Date
        @date={{this.date}}
      />
    `);

    await click('[data-test-field="Date"]');
-     await waitUntilCalendarAppears(); // a test helper that uses waitUntil()
    const dates = getAvailableDates();
    await click(dates[3]);

    assert.dom('[data-test-field="Date"]').hasValue('09.08.2024');
  });
});
```

Reply:
Thank you for that explanation!